### PR TITLE
RUBY-2138 background thread leak checker


### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/RUBY-2138

### CODE REVIEW

File: spec/integration/check_clean_slate_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'lite_spec_helper'
+
+# This test can be used to manually verify that there are no leaked
+# background threads - execute it after executing another test (in the same
+# rspec run) that is suspected to leak background threads.
```
Comment (egiurleo) [inline spec/integration/check_clean_slate_spec.rb:5]: Could you provide some guidance for how to run this test in the same rspec run as another test?

Comment (p-mongo) [reply spec/integration/check_clean_slate_spec.rb:5]: It's literally just `rspec foo_spec.rb spec/integration/check_clean_slate_spec.rb`.  How would you like me to phrase that?

Comment (egiurleo) [reply spec/integration/check_clean_slate_spec.rb:5]: Oh that's great. I didn't know you could do that. Could you just include that in the comment?
```
execute it after executing another test in the same rspec run with the following command:

rspec your_spec.rb spec/integration/check_clean_slate_spec.rb
```

Does that sound good?

Comment (p-mongo) [reply spec/integration/check_clean_slate_spec.rb]: Yup, added to the comment thanks.

Comment (egiurleo) [reply spec/integration/check_clean_slate_spec.rb]: Thanks!

**Pull request merged**